modules/RTC/RTCUtils.js

```diff
@@ -1144,11 +1038,34 @@ class RTCUtils extends Listenable {
 
             this._initPCConstraints(options);
 
-            // Call onReady() if Temasys plugin is not used
-            if (!browser.isTemasysPluginUsed()) {
-                onReady(options, this.getUserMediaWithConstraints.bind(this));
-                resolve();
+            rtcReady = true;
+            eventEmitter.emit(RTCEvents.RTC_READY, true);
+            screenObtainer.init(
+                options, this.getUserMediaWithConstraints.bind(this));
+
+            if (this.isDeviceListAvailable()
+                    && rawEnumerateDevicesWithCallback) {
+                rawEnumerateDevicesWithCallback(ds => {
+                    availableDevices = ds.splice(0);
+
+                    logger.debug('Available devices: ', availableDevices);
+                    sendDeviceListToAnalytics(availableDevices);
+
+                    eventEmitter.emit(RTCEvents.DEVICE_LIST_AVAILABLE,
+                        availableDevices);
+
+                    if (isDeviceChangeEventSupported) {
+                        navigator.mediaDevices.addEventListener(
+                            'devicechange',
+                            () => this.enumerateDevices(
+                                    onMediaDevicesListChanged));
+                    } else {
+                        pollForAvailableMediaDevices();
+                    }
+                });
             }
+
+            resolve();
```

Comment:
have you looked if it's possible to change the init method to not be a promise  (can be done in separate commit/PR if you want to look into that) ?

Reply:
I thought of it. While this would indeed simplify some stuff we do today, I worry we might need the process to be async in the future.

So I'd let it be as-is for a bit and revisit.